### DISCUSSION TITLE
What does the `(!)` mean in the output of `poetry show`

### DISCUSSION THREAD

**Discussion by KotlinIsland** [python-poetry/poetry#5040] [Q&A] [answered]
Poetry show puts a `(!)` before the version number of some packages and I can't figure out what it means.

`poetry help show` doesn't mention anything and I can't find anything in the docs about it.

```
> poetry show
safety                     (!) 1.10.3    Checks installed dependencies for known vulnerabilities.
```

**Comment by elliot-100** [comment, marked as answer]
Looking at the source, it means 'not installed', as a fallback to showing the package name in red:

https://github.com/python-poetry/poetry/blob/81c7d00cb3985c6db81d8370ef400731b05bf047/src/poetry/console/commands/show.py#L308-L313

See also #6330 and my related question: #7311
